packages/forms/test/form_group_spec.ts

```diff
@@ -86,6 +86,67 @@ import {of } from 'rxjs';
 
     });
 
+    describe('markAllAsTouched', () => {
+      it('should mark all descendants as touched', () => {
+        const formGroup: FormGroup = new FormGroup({
+          'c1': new FormControl('v1'),
+          'group': new FormGroup({'c2': new FormControl('v2'), 'c3': new FormControl('v3')}),
+          'array': new FormArray([
+            new FormControl('v4'), new FormControl('v5'),
+            new FormGroup({'c4': new FormControl('v4')})
+          ])
+        });
+
+        expect(formGroup.touched).toBe(false);
+
+        const control1 = formGroup.get('c1') as FormControl;
+
+        expect(control1.touched).toBe(false);
+
+        const innerGroup = formGroup.get('group') as FormGroup;
+
+        expect(innerGroup.touched).toBe(false);
+
+        const innerGroupFirstChildCtrl = innerGroup.get('c2') as FormControl;
+
+        expect(innerGroupFirstChildCtrl.touched).toBe(false);
+
+        formGroup.markAllAsTouched();
+
+        expect(formGroup.touched).toBe(true);
+
+        expect(control1.touched).toBe(true);
+
+        expect(innerGroup.touched).toBe(true);
+
+        expect(innerGroupFirstChildCtrl.touched).toBe(true);
+
+        const innerGroupSecondChildCtrl = innerGroup.get('c3') as FormControl;
+
+        expect(innerGroupSecondChildCtrl.touched).toBe(true);
+
+        const array = formGroup.get('array') as FormArray;
+
+        expect(array.touched).toBe(true);
+
+        const arrayFirstChildCtrl = array.controls[0] as FormControl;
```

Comment:
More idiomatic to use the public API method instead of grabbing from the `controls` array:

```suggestion
        const arrayFirstChildCtrl = array.at(0) as FormControl;
```

Same below